### PR TITLE
Update scripting examples dependency links and package layout

### PR DESCRIPTION
**Pull request by Karl-G1**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/niveristand-data-sharing-framework-custom-device/blob/master/CONTRIBUTING.md).

### What does this Pull Request accomplish?

- Create a second package for DSF that installs the scripting API and examples
- Add post-build VI to scripting API build that moves files to installed directory after build
- Fix linking problems inside the scripting examples that currently point to VIs in source locations
   - accomplished this by linking the VI dependencies in a directory without the source VIs in the hierarchy
   - also required excluding built libraries in the scripting examples build
- Move scripting examples library into correct directory in source
- Move scripting tests out of the main custom device project 
   - need to create a bug to track adding these back in if desired
   - having the tests in the project forced the example VIs to link to the source API

### Why should this Pull Request be merged?

The current install behavior for DSF includes VeriStand and LabVIEW support in one package. This should be split into two packages so that LabVIEW support isn't required for the Custom Device to install.

Additionally, the scripting example VIs all have broken dependencies when built and moved to the install location. This fixes the broken dependencies by only linking to VIs in the installed location.

### What testing has been done?

Lots of manual testing around the scripting API example changes.
PR Build validated that the new packages and built scripting examples link correctly.

### CODE REVIEW

File: build.toml [hunk 1]
```diff
@@ -117,13 +117,25 @@ target = 'Linux RT x64'
...
+[[package]]
 type = 'nipkg'
 control_file = 'control_dsf_cd'
 package_output_dir = 'Built'
 
 [package.payload_map]
 'Built\\Custom Device' = 'documents\National Instruments\NI VeriStand {veristand_version}\Custom Devices\Data Sharing Framework'
-'Built\\Scripting API' = 'ni-paths-LV{veristand_version}DIR\vi.lib\addons\VeriStand Custom Device Scripting APIs\Data Sharing Framework'
-'Built\\Scripting Examples' = 'ni-paths-LV{veristand_version}DIR\examples\NI VeriStand Custom Devices\Data Sharing Framework'
 'Built\\Errors' = 'ni-paths-NISHAREDDIR\Errors\English'
```
Comment (rtzoeller) [inline build.toml:127]: My gut feel is that the custom device should be installing an errors file to the year-version specific runtime directory `ni-paths-NISHAREDDIR\LabVIEW Run-Time\{veristand_version}\errors\English`, and that the scripting API should be installing it to `ni-paths-NISHAREDDIR\Errors\English`.

@buckd thoughts?

File: control_dsf_examples [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+Description: Provides LabVIEW examples for the NI Data Sharing Framework Custom Device for NI VeriStand {veristand_version}.
+XB-Eula: eula-ni-standard
+Priority: standard
+Homepage: http://www.ni.com
+XB-LanguageSupport: en
+XB-UserVisible: yes
+Section: Add-Ons
+XB-DisplayVersion: {display_version}
+XB-DisplayName: NI Data Sharing Framework LabVIEW Support for VeriStand {veristand_version}
+Depends: ni-data-sharing-framework-veristand-{veristand_version}-labview-support (>= {display_version}), ni-data-sharing-framework-plugins-{veristand_version}-support (>= {display_version})
```
Comment (rtzoeller) [inline control_dsf_examples:15]:
```suggestion
Depends: ni-data-sharing-framework-veristand-{veristand_version}-labview-support (>= {display_version}), ni-data-sharing-framework-plugins-veristand-{veristand_version}-support (>= {display_version})
```

File: build.toml (hunk 1 above)
Comment (dbendele) [reply build.toml:127]: we currently have two distinct error files: runtime errors and scripting errors

Comment (rtzoeller) [reply build.toml:127]: How do these two distinct files get installed? Presumably to distinct locations?

Comment (dbendele) [reply build.toml:127]: in this repo:
'<Build>\<LV version>\Errors\DSF-errors.txt' mapped to 'ni-paths-NISHAREDDIR\Errors\English' 

in the plugins repo:
<Build>\<LV Version>\Errors\DSF-GE-ReflectiveMemory-errors.txt' mapped to 'ni-paths-NISHAREDDIR\LabVIEW Run-Time\{veristand_version}\errors\English'

on the plugins error file, we have previously noted the issue with installation of the (same) error file to a version-agnostic folder through different, version-specific packages .

Comment (rtzoeller) [reply build.toml:127]: I see - are there error codes that the custom device itself generates, which don't come from the plugins?

**Review by dbendele — APPROVED** [review]
changes look good and testing showed working examples with no searching and no dirty dots.

File: build.toml (hunk 1 above)
Comment (rtzoeller) [reply build.toml:127]: I'm fine moving this discussion off of this PR - it's not worth gating this obvious improvement on a mostly philosophical discussion on how we should install error files.

I'll follow up asynchronously.

Comment (dbendele) [reply build.toml:127]: I think Core returns LV errors. As we widen the yellow-brick road, we will likely create DSF-specific error codes with more context-specific description.

**Review by rtzoeller — APPROVED** [review]
Approving the package build changes; I did not examine the LV files.

**Review by niphilj — APPROVED** [review]
- [ ] Post-Build Action.vi
I'm a little baffled by this code so adding a comment with the reasoning would be helpful in a future commit. Is reading the name for debugging purpose? And why clean the error, then auto-index it out (instead of a simple tunnel) which could do away with the merge errors.
image

**Pull request merged**